Source/tusdotnet/Interfaces/ITusPipelineStore.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
```

Comment:
I'm not a fan of having this extra interface as it makes the code harder to understand and that it requires two different implementations in the store, depending on framework/ASP.NET pipeline. I think a better approach would be to extend ClientDisconnectGuardedReadOnlyStream to include the pipereader and have it being convertible to the pipereader, e.g. allow something like this in the DiskStore:

```
public async Task<long> AppendDataAsync(string fileId, Stream stream, CancellationToken cancellationToken)
{
    var pipeReader = stream.GetPipeReader() ?? PipeReader.Create(stream);

    // Rest of code would be the same and it would work from tests
}
```